core/modules/filters/getvariablejson.js

```diff
@@ -0,0 +1,52 @@
+/*\
+title: $:/core/modules/filters/getvariablejson.js
+type: application/javascript
+module-type: filteroperator
+
+Filter operator to get widget variable info and
+Display as JSON with basic formatting
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+/*
+Export our filter function
+*/
+exports.getvariablejson = function(source,operator,options) {
+	var results = [],
+		widget = options.widget;
+	// "replacer" must be defined, otherwise JSON.stringify will throw a circular reference error RSOD
+	var replacer= "params name value default text isFunctionDefinition isProcedureDefinition isWidgetDefinition isMacroDefinition isVariable".split(" ");
+	source(function(tiddler,title) {
+		var text,
+			out = {},
+			info = widget.getVariableInfo(title, {});
+
+		out.text = info.text;
+		if (info.params) {
+			// unify parameter elements
+			out.params = info.params.map(function(param) {
+				return {"name":param.name, "default": param.value || param.default || ""}
+			});
+		};
+		if (info.srcVariable) {
+			out.value = info.srcVariable.value;
+			out.isFunctionDefinition = (info.srcVariable.isFunctionDefinition) ? "yes" : "no";
+			out.isProcedureDefinition = (info.srcVariable.isProcedureDefinition) ? "yes" : "no";
+			out.isWidgetDefinition = (info.srcVariable.isWidgetDefinition) ? "yes" : "no";
+			out.isMacroDefinition = (info.srcVariable.isMacroDefinition) ? "yes" : "no";
+			out.isVariable = (info.srcVariable.isFunctionDefinition ||
```

Comment:
The name "isVariable" does not make sense; functions, widgets, macros and procedures are all variables, and so it would be confusing for "isVariable" to equal "no".

Perhaps "isPlain" or "isSimple"?

Or maybe we don't need this at all? What is the use case?

If the intent is to be able to distinguish plain variables set with the `<$set>` widget from definitions, then things will get complicated when/if we extend the set widget to be able to define procedures/widgets/functions.

On balance, I think we should remove this.